Makefile

```diff
@@ -1,20 +1,35 @@
 all: kops
 
-DOCKER_REGISTRY=gcr.io/must-override/
-S3_BUCKET=s3://must-override/
+DOCKER_REGISTRY?=gcr.io/must-override/
```

Comment:
http://stackoverflow.com/questions/2826029/passing-additional-variables-from-command-line-to-make .. this lets you pick up the variables from the environment, rather than having to override it on the make command line. The latter also doesn't pass down through submakes and such, so it can be a pain sometimes.